### DISCUSSION TITLE
digital certificate (x.509) using the ML-DSA key pair in Python

### DISCUSSION THREAD

**Discussion by Aswin2203** [open-quantum-safe/liboqs#1956] [General]
I am trying to generate a digital certificate (x.509) using the ML-DSA key pair, but I don't have a support to that in python as I am using liboqs-python is there any possible ways to generate a x.509 certificate using ML-DSA key pair as it is the only possible way that can support pki infrastructure.

**Comment by SWilson4** [comment]
Hi @Aswin2203, please note that liboqs doesn't yet support the FIPS 204 final version of ML-DSA. (Only the "IPD" version.)

I don't believe there is an actively supported way to generate X509 certificates with liboqs-python, although [a contributor is working on it](https://github.com/open-quantum-safe/liboqs-python/pull/94)—perhaps you could take a look at that PR to see if it would suit your needs (and possibly contribute).
